test/arcade/patch/agent/cell/PatchCellCARTCD4Test.java

```diff
@@ -0,0 +1,454 @@
+package arcade.patch.agent.cell;
+
+import java.lang.reflect.Field;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import sim.engine.Schedule;
+import sim.engine.Steppable;
+import ec.util.MersenneTwisterFast;
+import arcade.core.sim.Simulation;
+import arcade.core.util.MiniBox;
+import arcade.core.util.Parameters;
+import arcade.patch.agent.module.PatchModule;
+import arcade.patch.agent.process.PatchProcessInflammation;
+import arcade.patch.agent.process.PatchProcessMetabolism;
+import arcade.patch.agent.process.PatchProcessSignaling;
+import arcade.patch.env.location.PatchLocation;
+import arcade.patch.sim.PatchSimulation;
+import arcade.patch.util.PatchEnums.Domain;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static arcade.core.ARCADETestUtilities.randomDoubleBetween;
+import static arcade.core.ARCADETestUtilities.randomIntBetween;
+import static arcade.patch.util.PatchEnums.AntigenFlag;
+import static arcade.patch.util.PatchEnums.State;
+
+public class PatchCellCARTCD4Test {
+
+    private Parameters parametersMock;
+
+    private PatchLocation locationMock;
+
+    private PatchCellContainer container;
+
+    private PatchCellCARTCD4 cellMock;
+
+    private PatchSimulation simMock;
+
+    @BeforeEach
+    public final void setUp() {
+        parametersMock = spy(new Parameters(new MiniBox(), null, null));
+        locationMock = mock(PatchLocation.class);
+
+        int id = 1;
+        int parentId = 1;
+        int pop = 1;
+        int age = randomIntBetween(1, 120950);
```

Comment:
I would just set this to a value between 0, 100, and then you can set the apoptosis age to something less specific and more intuitive